executor/show_test.go

```diff
@@ -369,6 +369,16 @@ func (s *testSuite) TestShow(c *C) {
 			"  `x` int(11) DEFAULT NULL,\n"+
 			"  PRIMARY KEY (`y`)\n"+
 			") ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin"))
+
+	// Test show create table with zerofill flag
+	tk.MustExec(`drop table if exists t`)
+	tk.MustExec(`create table t(id int primary key, val tinyint(10) zerofill);`)
```

Comment:
Add a year type?

Reply:
year with zerofill?

Reply:
It is not allowed, according to https://github.com/pingcap/tidb/pull/7542#issuecomment-416974493 @shenli